test/firtool/memoryMask.fir

```diff
@@ -0,0 +1,41 @@
+; RUN: firtool %s --format=fir | FileCheck %s  --check-prefixes COMMON,NOLOWER
+; RUN: firtool %s --format=fir -lower-memories | FileCheck %s --check-prefixes COMMON,LOWER
+
+circuit ByteEnableMemory:
```

Comment:
Nit: changing the name of the circuit to what the file is testing would be useful. Also, the name of the file would be better as `lower-memories.fir` or something like that.